### PR TITLE
ROE-2418 Add ids to Trust Individuals URL

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2418

### Change description

* Ids now present in URL when moving from Trust Details to Individuals screen
* New tests added

### Work checklist

- [X] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -443,3 +443,5 @@ export const PAYMENT_FAILED_WITH_PARAMS_URL = REGISTER_AN_OVERSEAS_ENTITY_WITH_P
 export const BENEFICIAL_OWNER_TYPE_SUBMIT_WITH_PARAMS_URL = REGISTER_AN_OVERSEAS_ENTITY_WITH_PARAMS_URL + BENEFICIAL_OWNER_TYPE_PAGE + SUBMIT_URL;
 export const PAYMENT_WITH_TRANSACTION_WITH_PARAMS_URL = REGISTER_AN_OVERSEAS_ENTITY_WITH_PARAMS_URL + PAYMENT;
 
+// TODO This will be in conflict when Mohammed's changes are merged. It will need removing
```
Comment (mwestacott) [inline src/config/index.ts:446]: See TODO comment. Will go later.

File: src/controllers/trust.details.controller.ts [hunk 2]
```diff
@@ -6,6 +6,6 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
   getTrustDetails(req, res, next, false, false);
 };
 
-export const post = (req: Request, res: Response, next: NextFunction) => {
-  postTrustDetails(req, res, next, false, false);
+export const post = async (req: Request, res: Response, next: NextFunction) => {
```
Comment (mwestacott) [inline src/controllers/trust.details.controller.ts:9]: Identified that this was missing during testing. Function should be asynchronous and awaiting the response from `postTrustDetails`.

File: test/controllers/trust.details.controller.spec.ts [hunk 3]
```diff
@@ -255,8 +278,143 @@ describe('Trust Details controller', () => {
...
+
+    test('catch error when post data from page', () => {
+      const error = new Error(ANY_MESSAGE_ERROR);
+      mockGetApplicationData.mockImplementationOnce(() => {
+        throw error;
+      });
+
       post(mockReq, mockRes, mockNext);
 
+      expect(mockNext).toBeCalledTimes(1);
```
Comment (markpit) [inline test/controllers/trust.details.controller.spec.ts:326]: should the post on line 326 also have the await?

Comment (mwestacott) [reply test/controllers/trust.details.controller.spec.ts:326]: Yes, probably. Will add.

File: test/controllers/trust.details.controller.spec.ts [hunk 4]
```diff
@@ -255,8 +278,143 @@ describe('Trust Details controller', () => {
...
+      expect(mockNext).toBeCalledWith(error);
+    });
+
+    test('add new trust in session', async () => {
+      mockIsActiveFeature.mockReturnValueOnce(false);
```
Comment (mattch1) [inline test/controllers/trust.details.controller.spec.ts:331]: Can we get a comment here as well?

File: test/controllers/trust.details.controller.spec.ts [hunk 5]
```diff
@@ -255,8 +278,143 @@ describe('Trust Details controller', () => {
...
+
+      expect(mockRes.redirect).toBeCalledTimes(1);
+      expect(mockRes.redirect).toBeCalledWith(NEXT_PAGE_URL);
+
+      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
+      expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(`${TRUST_ENTRY_WITH_PARAMS_URL}/dummyId${TRUST_INVOLVED_URL}`);
+    });
+
+    test('update existing trust in session', async () => {
+      mockIsActiveFeature.mockReturnValueOnce(false);
```
Comment (mattch1) [inline test/controllers/trust.details.controller.spec.ts:399]: As above

File: test/controllers/trust.details.controller.spec.ts (hunk 3 above)
Comment (mwestacott) [reply test/controllers/trust.details.controller.spec.ts:326]: Done.

File: test/controllers/trust.details.controller.spec.ts (hunk 4 above)
Comment (mwestacott) [reply test/controllers/trust.details.controller.spec.ts]: Done.

File: test/controllers/trust.details.controller.spec.ts (hunk 5 above)
Comment (mwestacott) [reply test/controllers/trust.details.controller.spec.ts]: Done.

**Review by mattch1 — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**